app/controllers/concerns/find_common.rb

```diff
@@ -54,4 +54,8 @@ def resource_class_for(resource)
       return nil
     end
   end
+
+  def find_hosts_count
+    @hosts_count = Host::Managed.reorder('').authorized.group("#{resource_name}_id").count
+  end
```

Comment:
I understand but it leaves space for errors in the future. Explicit "pull" mechanism might be safer.